sklearn/datasets/_samples_generator.py

```diff
@@ -306,14 +328,54 @@ def make_classification(
         scale = 1 + 100 * generator.uniform(size=n_features)
     X *= scale
 
+    indices = np.arange(n_features)
     if shuffle:
         # Randomly permute samples
         X, y = util_shuffle(X, y, random_state=generator)
 
         # Randomly permute features
-        indices = np.arange(n_features)
         generator.shuffle(indices)
-        X[:, :] = X[:, indices]
+
+    X[:, :] = X[:, indices]
```

Comment:
this is introducing a change which we don't need / want to have. This should only happen if `shuffle` is `True`.